validation_tests/superlu-dist-nonspacktest/EXAMPLE/compile.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+#. ../setup.sh
+#mpicc 
+${TEST_CC_MPI} \
+ -I${SUPERLU_DIST_ROOT}/include  \
+-DUSE_VENDOR_BLAS -fopenmp -std=c99 -O3 -DPRNTlevel=1 -DDEBUGlevel=0 -fPIE \
+-o pddrive.c.o -c pddrive.c
+
+#mpicc 
+${TEST_CC_MPI} \
+ -I${SUPERLU_DIST_ROOT}/include \
+-DUSE_VENDOR_BLAS -fopenmp -std=c99 -O3 -DPRNTlevel=1 -DDEBUGlevel=0 -fPIE \
+-o dcreate_matrix.c.o -c dcreate_matrix.c
+
+#mpicxx 
+${TEST_CXX_MPI} \
+-fopenmp -std=c++11 -rdynamic -O3 \
+pddrive.c.o dcreate_matrix.c.o -o pddrive \
+${SUPERLU_DIST_ROOT}/lib/libsuperlu_dist.a \
+/opt/cray/pe/libsci/21.08.1.2/GNU/9.1/x86_64/lib/libsci_gnu_82_mpi_mp.a \
```

Comment:
Can this be replaced with something generic that will allow this test to pass on other platforms? <PACKAGE>_ROOT variables are created for all dependencies if that helps. I would rather avoid them when possible but if we need platform specific variables they can be defined in settings.sh in the top-level testsuite directory.

Reply:
The issue is that the blas dependency of superlu-dist for different e4s versions are quite different. For example, 
on Cori, 
module load e4s/21.05
spack find --deps superlu-dist
gives 
-- cray-cnl7-haswell / intel@19.1.3.304 -------------------------
superlu-dist@6.4.0
    metis@5.1.0
    mpich@3.1
    openblas@0.3.10
    parmetis@4.0.3

but on Perlmutter,
module load e4s/21.11-tcl
spack find --deps superlu-dist
gives 
-- cray-sles15-zen2 / gcc@9.3.0 ---------------------------------
superlu-dist@7.1.1
    cray-libsci@21.08.1.2
    cray-mpich@8.1.12
    metis@5.1.0
    parmetis@4.0.3

Is there an automatic way of detecting the blas dependency of a given spack superlu-dist installation? Also, there is OPENBLAS_ROOT for e4s/21.05, but there is no such thing for libsci for e4s/21.11-tcl, as it's using Perlmutter's system libsci.